Python import error due to docstring formatting issue

Hi,

When running:
```python
from proxsuite.torch.qplayer import QPFunction
```
I sometimes get the following error:
```
E     File "/home/runner/work/TorchJD/TorchJD/.venv/lib/python3.10/site-packages/cmeel.prefix/lib/python3.10/site-packages/proxsuite/torch/qplayer.py", line 21
E       """!
E       ^^^
E   SyntaxError: invalid escape sequence '\m'
```

It seems that some python docstrings are incorrectly formatted (i.e. missing the `r` specifier to make it a raw docstring, when it contains a backslash).
I can't reproduce in my local env (the error may be silent in some cases, or depend on the python version or something), but in most of my CI's environments it breaks, and looking at the code, I can confirm that the docstring is of the form `""" docstring """` rather than `r""" docstring """`.

I can make a PR to fix this. Also note that this can be statically checked by ruff W605 (https://docs.astral.sh/ruff/rules/invalid-escape-sequence/), if you wanna prevent this from happening again.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - replace `std::numeric_limits<T>::infinity()` by `std::numeric_limits<T>::max()` ([#413](https://github.com/Simple-Robotics/proxsuite/pull/413))
 - Upgraded nanobind dependency version (submodule) to v2.9.2 ([#418](https://github.com/Simple-Robotics/proxsuite/pull/418))
 - Better dynamic module handling ([#419](https://github.com/Simple-Robotics/proxsuite/pull/419))
+- Fix `SyntaxError` on importing `proxsuite.torch.qplayer` caused by invalid escape sequences in the `QPFunction` docstring ([#458](https://github.com/Simple-Robotics/proxsuite/pull/461))
 
 ### Fixed
 - Correction of the status update for `PROXQP_SOLVED_CLOSEST_PRIMAL_FEASIBLE` ([#432](https://github.com/Simple-Robotics/proxsuite/pull/432))

--- a/bindings/python/proxsuite/torch/qplayer.py
+++ b/bindings/python/proxsuite/torch/qplayer.py
@@ -18,7 +18,7 @@ def QPFunction(
     omp_parallel=False,
     structural_feasibility=True,
 ):
-    """!
+    r"""!
     Solve a batch of Quadratic Programming (QP) problems.
 
     This function solves QP problems of the form:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,4 +25,5 @@ configure-args = ["-DBUILD_TESTING:BOOL=OFF","-DBUILD_PYTHON_INTERFACE:BOOL=ON",
 
 [tool.ruff.lint]
 ignore = [ "E741" ]
+extend-select = ["W605"]
 exclude = [ "cmake-module/*", "bindings/python/external/nanobind/*" ]

